piff/gsobject_model.py

```diff
@@ -0,0 +1,341 @@
+        attributes.  This is a single-step solution if only solving for flux,
+        otherwise an iterative operation.  DOF in the result assume
+        only flux (& center) are free parameters.
+
+        :param star:        A Star instance
+        :param fit_center:  If False, disable any motion of center
+        :param logger:      A logger object for logging debug info. [default: None]
+
+        :returns:           New Star instance, with updated flux, center, chisq, dof, worst
+        """
+        if logger:
+            logger.debug("Reflux for star:")
+            logger.debug("    flux = %s",star.fit.flux)
+            logger.debug("    center = %s",star.fit.center)
+            logger.debug("    props = %s",star.data.properties)
+            logger.debug("    image = %s",star.data.image)
+            #logger.debug("    image = %s",star.data.image.array)
+            #logger.debug("    weight = %s",star.data.weight.array)
+            logger.debug("    image center = %s",star.data.image(star.data.image.center()))
+            logger.debug("    weight center = %s",star.data.weight(star.data.weight.center()))
+        do_center = fit_center and self._force_model_center
+        if do_center:
+            params = self._lmfit_params(star, vary_params=False)
+            results = self._lmfit_minimize(params, star, logger=logger)
+            return Star(star.data, StarFit(star.fit.params,
+                                           flux = results.params['flux'].value,
+                                           center = (results.params['cenu'].value,
+                                                     results.params['cenv'].value),
+                                           chisq = results.chisqr,
+                                           dof = np.count_nonzero(star.data.weight.array) - 3,
+                                           alpha = star.fit.alpha,
+                                           beta = star.fit.beta))
+        else:
+            image, weight, image_pos = star.data.getImage()
+            model_image = self.draw(star).image
+            flux_ratio = (np.sum(weight.array * image.array * model_image.array)
+                          / np.sum(weight.array * model_image.array**2))
+            new_chisq = np.sum(weight.array * (image.array - flux_ratio*model_image.array)**2)
+            return Star(star.data, StarFit(star.fit.params,
+                                           flux = star.flux*flux_ratio,
+                                           center = star.fit.center,
+                                           chisq = new_chisq,
+                                           dof = np.count_nonzero(weight.array) - 1,
+                                           alpha = star.fit.alpha,
+                                           beta = star.fit.beta))
+
+
+# Wrapped Gaussian model for backwards compatibility.
+# Should we do this for Kolmogorov and Moffat too?
```

Comment:
I now have named versions as well as the eval/repr trick, and tests for all.  :)